src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperCompletionFactsService.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    [Export(typeof(TagHelperCompletionFactsService))]
+    internal class DefaultTagHelperCompletionFactsService : TagHelperCompletionFactsService
+    {
+        [Import(typeof(TagHelperFactsService))]
+        public TagHelperFactsService TagHelperFactsService { get; set; }
+
+        public override ElementCompletionResult GetElementCompletions(
```

Comment:
Yup, doing it in two check-ins. One for element completion, one for attribute completion.